rse-bash.Rmd

```diff
@@ -0,0 +1,3902 @@
+```
+
+(Again, your results may be slightly different depending on your operating
+system and how you have customized your filesystem.)
+
+`ls` prints the names of the files and directories in the current directory.
+We can make its output more comprehensible by using the `-F` **option**
+(also known as a **switch** or a **flag**) ,
+which tells `ls` to classify the output
+by adding a marker to file and directory names to indicate what they are:
+- a trailing `/` indicates that this is a directory
+- `@` indicates a link
+- `*` indicates an executable
+
+Depending on your default options,
+the shell might also use colors to indicate whether each entry is a file or
+directory.
+
+```shell
+$ ls -F
+```
+
+```text
+Applications/ Documents/    Library/      Music/        Public/
+Desktop/      Downloads/    Movies/       Pictures/
+```
+
+Here,
+we can see that our home directory contains mostly **sub-directories**.
+Any names in your output that don't have a classification symbol
+are plain old **files**.
+
+
+### General syntax of a shell command
+
+Consider the command below as a general example of a command,
+which we will dissect into its component parts:
+
+```shell
+$ ls -F /
+```
+
+
+`ls` is the **command**, with an **option** `-F` and an
+**argument** `/`.
+We've already encountered options (also called **switches** or **flags**) which
+either start with a single dash (`-`) or two dashes (`--`), and they change the behaviour of a command.
+Arguments tell the command what to operate on (e.g. files and directories).
+Sometimes options and arguments are referred to as **parameters**.
+A command can be called with more than one option and more than one argument: but a
+command doesn't always require an argument or an option.
+
+Each part is separated by spaces: if you omit the space
+between `ls` and `-F` the shell will look for a command called `ls-F`, which
+doesn't exist. Also, capitalization can be important: `ls -r` is different to `ls -R`.
+
+Putting all that together, our command above gives us a listing
+of files and directories in the root directory `/`.
+An example of the output you might get from the above command is given below:
+
+```shell
+$ ls -F /
+Applications/         System/
+Library/              Users/
+Network/              Volumes/
+```
+
+### Getting help
+
+`ls` has lots of other **options**. There are two common ways to find out how
+to use a command and what options it accepts:
+
+1. We can pass a `--help` option to the command, such as: `$ ls --help`
+
+2. We can read its manual with `man`, such as: `$ man ls`
+
+Depending on your environment you might find that only one of these works
+(either `man` or `--help`).
+We'll describe both ways below.
+
+
+#### The `--help` option
+
+Many bash commands, and programs that people have written that can be
+run from within bash, support a `--help` option to display more
+information on how to use the command or program.
+
+```shell
+$ ls --help
+```
+
+```text
```

Comment:
This is waaay too much - is there a simpler man page we can show?

Reply:
`ls` is basically the only command that has been introduced at that point, so the only option would be to introduce help/man later on... I'm not sure whether that's a good idea or not.